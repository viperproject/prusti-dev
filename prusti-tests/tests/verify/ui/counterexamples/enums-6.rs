// compile-flags: -Pcounterexample=true

use prusti_contracts::*;

enum BinOp {
    Add(i32, i32),
    Sub(i32, i32),
    Div(i32, i32),
}

fn apply(op: BinOp) -> i32 {
    match op {
        BinOp::Add(a, b) => a + b,
        BinOp::Sub(c, d) => c - d,
        BinOp::Div(e, f) => {
            if e == 1 { // force specific counterexample
                e / f
            } else {
                0
            }
        }
    }
}

fn main() {}
