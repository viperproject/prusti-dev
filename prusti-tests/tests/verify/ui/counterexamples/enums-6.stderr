error: [Prusti: verification error] assertion might fail with "attempt to divide by zero"
  --> $DIR/enums-6.rs:17:17
   |
17 |                 e / f
   |                 ^^^^^
   |
note: counterexample for "op"
        initial value: BinOp::Div(
            1,
            0,
        )
        final value:   BinOp::Div(
            1,
            0,
        )
  --> $DIR/enums-6.rs:11:10
   |
11 | fn apply(op: BinOp) -> i32 {
   |          ^^
note: counterexample for "a"
        final value:   0
  --> $DIR/enums-6.rs:13:20
   |
13 |         BinOp::Add(a, b) => a + b,
   |                    ^
note: counterexample for "b"
        final value:   0
  --> $DIR/enums-6.rs:13:23
   |
13 |         BinOp::Add(a, b) => a + b,
   |                       ^
note: counterexample for "c"
        final value:   0
  --> $DIR/enums-6.rs:14:20
   |
14 |         BinOp::Sub(c, d) => c - d,
   |                    ^
note: counterexample for "d"
        final value:   0
  --> $DIR/enums-6.rs:14:23
   |
14 |         BinOp::Sub(c, d) => c - d,
   |                       ^
note: counterexample for "e"
        final value:   1
  --> $DIR/enums-6.rs:15:20
   |
15 |         BinOp::Div(e, f) => {
   |                    ^
note: counterexample for "f"
        final value:   0
  --> $DIR/enums-6.rs:15:23
   |
15 |         BinOp::Div(e, f) => {
   |                       ^
note: counterexample for result
        final value:   ?
  --> $DIR/enums-6.rs:11:24
   |
11 | fn apply(op: BinOp) -> i32 {
   |                        ^^^

error: aborting due to previous error

